RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterQuickFix.cs

```diff
@@ -47,17 +49,20 @@ public override void Fix()
 
             SetLocalCopyVariableName();
 
-            if (IsCancelled) { return; }
+            if (!ProposedLocalVariableNameIsValid()
+                || IsCancelled)
+            {
+                return;
+            }
 
             ModifyBlockToUseLocalCopyVariable();
         }
 
         private void SetLocalCopyVariableName()
         {
-            using (var view = new AssignedByValParameterQuickFixDialog(_originalProcCodeLines))
+            using (var view = new AssignedByValParameterQuickFixDialog(_target, Selection))
```

Comment:
For this exact reason, the AssignedByValParameterQuickFix has an accessor 'TESTONLY_FixUsingAutoGeneratedName(...)' [not on the QuickFix interface].  This function is there for the sole purpose of supporting unit tests.  This accessor calls "Fix()" after setting an internal flag that suppresses the call to 'ShowDialog()'.  Didn't really like adding the test-support method, but couldn't think of (or find an example) of a better way.